bank/api/middleware.go

```diff
@@ -0,0 +1,167 @@
+package api
+
+import (
+	"encoding/json"
+	"github.com/gorilla/context"
+	"github.com/iliyanmotovski/bankv2/bank/domain"
+	"log"
+	"net/http"
+	"time"
+)
+
+func CheckValidSessionOnLoginAndLogoutMiddleware(sessStore domain.SessionStore) func(http.Handler) http.Handler {
```

Comment:
Name should be more common as it may be used not just by login and logout: 

CookieBasedSecurity sounds better.